test/unit/test_ext_auth.py

```diff
@@ -0,0 +1,264 @@
+    ).is_false()
+    assert_that(
+        DCVAuthenticator.is_process_valid(_create_row(user, valid_command, "afsd"), user, session_id)
+    ).is_false()
+
+
+def mock_generate_random_token(mocker, value):
+    mocker.patch(MAIN + "generate_random_token", return_value=value)
+
+
+def mock_verify_session_existence(mocker, exists):
+    def _return_value(user, sessionid):
+        if not exists:
+            raise DCVAuthenticator.IncorrectRequestException("The given session for the user does not exists")
+
+    mocker.patch(EXT_AUTH + "_verify_session_existence", side_effect=_return_value)
+
+
+def mock_os(mocker, user, timestamp):
+    class FileProperty(object):
+        pass
+
+    file_property = FileProperty
+    file_property.st_uid = 1
+    file_property.st_size = b"800"
+    file_property.st_mtime = timestamp
+    file_property.pw_name = user
+    file_property.st_mode = 0
+
+    mocker.patch(MAIN + "os.stat", return_value=file_property, autospec=True)
+    mocker.patch(MAIN + "os.remove")
+    mocker.patch(MAIN + "getpwuid", autospec=True, return_value=file_property)
+
+
+def test_get_request_token_parameter():
+    with pytest.raises(DCVAuthenticator.IncorrectRequestException):
+        DCVAuthenticator._get_values_from_parameters(parameters={"a": "5", "b": "2"}, keys=["authUser", "sessionID"])
+    with pytest.raises(DCVAuthenticator.IncorrectRequestException):
+        DCVAuthenticator._get_values_from_parameters(
+            parameters={"authUser": "5", "b": "2"}, keys=["authUser", "sessionID"]
+        )
+    with pytest.raises(DCVAuthenticator.IncorrectRequestException):
+        DCVAuthenticator._get_values_from_parameters({"a": "5", "sessionID": "2"}, keys=["authUser", "sessionID"])
+    assert_that(
+        DCVAuthenticator._get_values_from_parameters(
+            {"authUser": "luca", "sessionID": "1234"}, keys=["authUser", "sessionID"]
+        )
+    ).is_equal_to(["luca", "1234"])
+    assert_that(
+        DCVAuthenticator._get_values_from_parameters({"requestToken": "ciao"}, keys=["requestToken"])
+    ).is_equal_to(["ciao"])
+
+
+def test_get_request_token(mocker):
+    token_value = "1234abcd_-"
+    user = "centos"
+    session_id = "mysession"
+
+    mock_verify_session_existence(mocker, exists=True)
+    mock_generate_random_token(mocker, token_value)
+    # all correct
+    assert_that(DCVAuthenticator._get_request_token(user, session_id)).is_equal_to(
+        json.dumps({"requestToken": token_value, "requiredFile": generate_sha512_hash(token_value)})
+    )
+    assert_that(DCVAuthenticator._request_token_manager.get_token_value(token_value)[:-1]).is_equal_to(
+        (user, session_id)
+    )
+    # session does not exists
+    mock_verify_session_existence(mocker, exists=False)
+    with pytest.raises(DCVAuthenticator.IncorrectRequestException):
+        DCVAuthenticator._get_request_token(user, session_id)
+
+
```

Comment:
Let's use https://docs.pytest.org/en/latest/parametrize.html also here and in the other tests when possible